common/neon/transaction_model.py

```diff
@@ -1,44 +1,70 @@
 from __future__ import annotations
 
-from typing import Union
+from typing import Union, Any
 
 from typing_extensions import Self
 
 from ..ethereum.bin_str import EthBinStr, EthBinStrField
 from ..ethereum.hash import EthTxHash, EthTxHashField, EthAddressField, EthAddress
 from ..ethereum.transaction import EthTx
-from ..utils.cached import cached_property, cached_method
+from ..utils.cached import cached_method
 from ..utils.format import str_fmt_object
 from ..utils.pydantic import BaseModel, HexUIntField
 
+from pydantic import Field
+
+_TX_MODEL_EXCLUDE_LIST= {
+    0: set(["max_priority_fee_per_gas", "max_fee_per_gas", "chain_id"]),
+    2: set(["gas_price"])
+}
 
 class NeonTxModel(BaseModel):
     tx_type: HexUIntField
+    chain_id: HexUIntField | None = Field(default=None)
     neon_tx_hash: EthTxHashField
     from_address: EthAddressField
     to_address: EthAddressField
     contract: EthAddressField
     nonce: HexUIntField
-    gas_price: HexUIntField
+    gas_price_legacy: HexUIntField | None = Field(default=None, serialization_alias="gas_price")
+    max_priority_fee_per_gas: HexUIntField | None = Field(default=None)
+    max_fee_per_gas: HexUIntField | None = Field(default=None)
     gas_limit: HexUIntField
     value: HexUIntField
     call_data: EthBinStrField
+    # Access List is missing, no support yet.
     v: HexUIntField
     r: HexUIntField
     s: HexUIntField
 
     error: str | None = None
 
+    def model_post_init(self, _ctx: Any) -> None:
+        if self.tx_type not in (0, 2):
+            raise ValueError("Unsupported transaction type.")
+        if self.tx_type == 0:
+            if self.gas_price_legacy is None:
+                raise ValueError("gas_price is not specified for the Legacy transaction.")
+            if self.max_priority_fee_per_gas is not None or self.max_fee_per_gas is not None:
+                raise ValueError("max_priority_fee_per_gas or max_fee_per_gas should be absent for the Legacy transaction.")
+        else:
+            if self.gas_price_legacy is not None:
+                raise ValueError("gas_price shouldn't be specified for the Dynamic Gas transaction.")
+            if self.max_priority_fee_per_gas is None or self.max_fee_per_gas is None:
+                raise ValueError("max_priority_fee_per_gas or max_fee_per_gas is absent for the Dynamic Gas transaction.")
+        return self
```

Comment:
remove 
`return self`
because the function doesn't return a value:
`def model_post_init(self, _ctx: Any) -> None:`

Reply:
done